### PR TITLE
Add email-finding tool and README

### PR DESCRIPTION
**Pull request by joekamibeppu**
This Pull Request contains three new files:
- `find_email_addresses.py`, the main program
- `README.md`
- `requirements.txt`

`README.md` contains instructions for installing dependencies and running the program. Thank you very much for reviewing my code.

### CODE REVIEW

File: find_email_addresses.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    emails += re.findall(email_pattern, response.text)
+    visited.append(url)
+
+    for link in BeautifulSoup(response.text, 'html.parser',
+                              parse_only=SoupStrainer('a')):
+        if link.has_attr('href'):
+            new_url = urljoin(url, link['href'])
+            if urlparse(new_url).hostname == urlparse(url).hostname:
+                if new_url not in visited:
+                    emails += find_emails(new_url, emails, visited)
```
Comment (julien-jana) [inline find_email_addresses.py]: Can you think of a way to limit the duration of the program? What if there's thousands of subpages found?

File: find_email_addresses.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import requests
+from bs4 import BeautifulSoup, SoupStrainer
+
+def find_emails(url, emails, visited):
+    ''' finds email addresses by recursively traversing through pages '''
+    #TODO: refine email regex pattern
+    email_pattern = r'\w[\w._-]*@[\w._-]*' + visited[0] # contains domain name
+    response = requests.get(url)
+    emails += re.findall(email_pattern, response.text)
+    visited.append(url)
```
Comment (julien-jana) [inline find_email_addresses.py]: Could you use a different data structure than a list here?

File: find_email_addresses.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    print "Found these email addresses:"
+    emails = set(emails)
+    for email in emails:
+        print email
+
+def main(domain):
+    ''' main function '''
+    emails = []
+    visited = [domain]     # preserve domain in first index
+    prefix = 'http://www.' # default is HTTP (some sites do not handle HTTPS)
```
Comment (julien-jana) [inline find_email_addresses.py]: Why add the www? Could this be breaking the web.mit.edu scrape?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,17 @@
 # email-finder
```
Comment (julien-jana) [inline README.md:1]: Nice documentation

**Review by julien-jana — COMMENTED** [review]
This Pull Requests currently has merge conflicts. Can you resolve those so it's ready to merge?

Also, can you think of a way to make mit.edu work?

**Comment by joekamibeppu** [comment]
Updates in this commit:
- The recursive crawling method was changed to an iterative method.
- All lists were changed to sets.
- If more than 500 subpages are found, the program stops and all emails collected at that point are printed out.

As written in the README, the program needs a full webpage name (eg. web.mit.edu instead of mit.edu) as an argument for its domain parameter.

**Pull request merged**